rvps/src/cache/local_fs/mod.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! This Cache stores RV information inside a local file
+
+use std::path::Path;
+
+use anyhow::*;
+
+use crate::{Cache, ReferenceValue};
+
+/// Local directory path to store the reference values,
+/// which is created by sled engine.
+const FILE_PATH: &str = "./reference_values";
+
+/// `LocalFs` implements [`Cache`] trait. And
+/// it uses rocksdb inside.
+pub struct LocalFs {
+    engine: sled::Db,
+}
+
+impl Default for LocalFs {
+    fn default() -> Self {
+        LocalFs::new(FILE_PATH).expect("Failed to create LocalFs Cache.")
+    }
+}
+
+impl LocalFs {
+    /// Create a new [`LocalFs`] with given
+    /// file storage path.
+    fn new(path: &str) -> Result<Self> {
+        let path = Path::new(path);
+        let engine = sled::open(path)?;
+        Ok(Self { engine })
+    }
+}
+
+impl Cache for LocalFs {
+    fn set(&mut self, name: String, rv: ReferenceValue) -> Result<Option<ReferenceValue>> {
+        let rv_serde = serde_json::to_vec(&rv)?;
+        let res = match self.engine.insert(name, rv_serde)? {
+            Some(v) => {
+                let v = serde_json::from_slice(&v)?;
+                Ok(Some(v))
+            }
+            None => Ok(None),
+        };
+
+        self.engine.flush()?;
+        res
+    }
+
+    fn get(&self, name: &str) -> Result<Option<ReferenceValue>> {
+        match self.engine.get(name)? {
+            Some(v) => {
+                let v = serde_json::from_slice(&v)?;
+                Ok(Some(v))
+            }
+            None => Ok(None),
+        }
```

Comment:
maybe replace the entire match with:

```rust
self.engine.get(name)?.map(|v| serde_json::from_slice(&v)?)
```

something similar can be done above in `set()`.

Up to you if it's clearer or you prefer the `match`.

Reply:
I got your idea, but Rust seems not able to support `?` in a `map` closure. I tried this, but failed

Reply:
Oh, you're right (it won't work because `?` works only when you return a Result, but the closure doesn't). Thanks.